File: zadanie2/D.py
def min_days_to_complete_tasks(n, k, directions):
    # Сортируем направления
    directions.sort()
    
    # Инициализация количества дней
    days = 0
    i = 0
    
    while i < n:
        # Увеличиваем счетчик дней
        days += 1
        
        # Устанавливаем предел для текущей группы дел
        current_end = directions[i] + k
        
        # Пропускаем все подходящие направления в текущий день
        while i < n and directions[i] <= current_end:
            i += 1
    
    return days

# Считывание данных
n, k = map(int, input().strip().split())
directions = list(map(int, input().strip().split()))

# Вывод результата
result = min_days_to_complete_tasks(n, k, directions)
print(result)

File: zadanie2/D.go
package main

import (
	"fmt"
	"math"
	"sort"
)

func minDaysToCompleteTasks(n int, k int, tasks []int) int {
	sort.Ints(tasks)
	days := 0
	i := 0
	for i < n {
		days++
		currentTask := tasks[i]
		for i < n && int(math.Abs(float64(tasks[i]-currentTask))) <= k {
			i++
		}
	}

	return days
}

func main() {
	var n, k int
	fmt.Scan(&n, &k)
	tasks := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&tasks[i])
	}
	result := minDaysToCompleteTasks(n, k, tasks)
	fmt.Println(result)
}
